### PR TITLE
pod package add pod resource and pod network funcs

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/pod/pod.go [hunk 1]
```diff
@@ -616,6 +616,83 @@ func (builder *Builder) WithSecondaryNetwork(network []*multus.NetworkSelectionE
...
+
+	glog.V(100).Infof("Applying SecurityContext configuration pod % in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if securityContext == nil {
+		glog.V(100).Infof("The 'securityContext' of the pod is empty")
+
+		builder.errorMsg = "'securityContext' parameter is empty"
+	}
+
```
Comment (trewest) [inline pkg/pod/pod.go:688]: check builder.errorMsg here

**Review by trewest — COMMENTED** [review]
Just one comment

File: pkg/pod/pod.go (hunk 1 above)
Comment (kononovn) [reply pkg/pod/pod.go:688]: We have this check under 	builder.isMutationAllowed("SecurityContext") function.

Comment (trewest) [reply pkg/pod/pod.go:688]: But you update the struct regardless. Is that okay?

Comment (kononovn) [reply pkg/pod/pod.go:688]: We will not update it if we fail on line 689

Comment (kononovn) [reply pkg/pod/pod.go:688]: sorry my bad. Fixed

**Review by trewest — DISMISSED** [review]
/lgtm

File: pkg/pod/pod.go [hunk 2]
```diff
@@ -616,6 +616,87 @@ func (builder *Builder) WithSecondaryNetwork(network []*multus.NetworkSelectionE
...
+	return builder
+}
+
+// WithSecurityContext sets SecurityContext on pod definition.
+func (builder *Builder) WithSecurityContext(securityContext *v1.PodSecurityContext) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying SecurityContext configuration pod % in namespace %s",
```
Comment (achuzhoy) [inline pkg/pod/pod.go:680]: Applying SecurityContext configuration on pod

Comment (achuzhoy) [reply pkg/pod/pod.go:680]: And also after pod should be %s

File: pkg/pod/container.go [hunk 3]
```diff
@@ -130,6 +132,109 @@ func (builder *ContainerBuilder) WithSecurityContext(securityContext *v1.Securit
...
+	if memory == "" {
+		glog.V(100).Infof("Container's resource request memory is empty")
+
+		builder.errorMsg = "container's resource request 'memory' is empty"
+	}
+
+	if cpu <= 0 {
+		glog.V(100).Infof("Container's resource request cpu can not be zero or negative number.")
+
+		builder.errorMsg = "container's resource request 'memory' is empty"
```
Comment (achuzhoy) [inline pkg/pod/container.go:191]: cpu not memory

File: pkg/pod/container.go [hunk 4]
```diff
@@ -130,6 +132,109 @@ func (builder *ContainerBuilder) WithSecurityContext(securityContext *v1.Securit
...
+	builder.definition.Resources.Requests = v1.ResourceList{
+		"hugepages-1Gi": resource.MustParse(hugePages),
+		"memory":        resource.MustParse(memory),
+		"cpu":           *resource.NewQuantity(int64(2), resource.DecimalSI),
+	}
+
+	return builder
+}
+
+// WithEnvVar add environment variables to container.
```
Comment (achuzhoy) [inline pkg/pod/container.go:207]: nit: adds

File: pkg/pod/pod.go [hunk 5]
```diff
@@ -616,6 +616,87 @@ func (builder *Builder) WithSecondaryNetwork(network []*multus.NetworkSelectionE
...
+	return builder
+}
+
+// WithHugePages sets hugePages on all containers inside the pod.
+func (builder *Builder) WithHugePages() *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying hugePages configuration to all pod's: %s containers", builder.Definition.Name)
```
Comment (achuzhoy) [inline pkg/pod/pod.go:641]: nit: Applying hugePages configuration to all containers in pod: %s

**Comment by kononovn** [comment]
@achuzhoy PTAL

File: pkg/pod/pod.go [hunk 6]
```diff
@@ -677,7 +677,7 @@ func (builder *Builder) WithSecurityContext(securityContext *v1.PodSecurityConte
 		return builder
 	}
 
-	glog.V(100).Infof("Applying SecurityContext configuration pod % in namespace %s",
+	glog.V(100).Infof("Applying SecurityContext configuration on pod % in namespace %s",
```
Comment (achuzhoy) [inline pkg/pod/pod.go:680]: after pod it's % - should be %s

**Pull request merged**